Tasks/DotNetCoreInstallerV0/dotnetcoreinstaller.ts

```diff
@@ -3,28 +3,80 @@ import * as toolLib from 'vsts-task-tool-lib/tool';
             throw tl.loc("ImplicitVersionNotSupported", version);
         }
+        if(!useGlobalJson && (version == null || version.length == 0)){
+            throw tl.loc("VersionMissingException", "If you don't use global.json then please defined a version.")
+        }
         this.version = version;
         this.cachedToolName = this.packageType === 'runtime' ? 'dncr' : 'dncs';;
     }
 
-    public async install() {
-        // Check cache
-        let toolPath: string;
-        let osSuffixes = this.detectMachineOS();
-        let parts = osSuffixes[0].split("-");
-        this.arch = parts.length > 1 ? parts[1] : "x64";
-        toolPath = this.getLocalTool();
+    private async installFromGlobalJson(osSuffixes: string[]) {
+        let filePathsToGlobalJson = this.getFiles(this.workingDirectory ? this.workingDirectory : ".", "global.json");
+        if(filePathsToGlobalJson.length == 0){
+            throw tl.loc("FailedToFindGlobalJson", "From the root of your working folder we can't find any global.json file.");
+        }
+        let sdkVersionNumber = new Array<{ name: string, toolPath?: string }>();
+        // read all global files
+        for (let index = 0; index < filePathsToGlobalJson.length; index++) {
+            const filePath = filePathsToGlobalJson[index];
+            let globalJson = {} as { sdk: { version: string } };  
+            console.log("global.json found here: " + filePath);
+            try {
+                globalJson = (JSON.parse(fileSystem.readFileSync(filePath).toString())) as { sdk: { version: string } };
+            } catch (error) {
+                throw tl.error("FailedToReadGlobalJson \r\n" + 
+                "The global.json file isn't valid. (" + filePath + ") Please take a look in the documentation for global.json (https://docs.microsoft.com/en-us/dotnet/core/tools/global-json)");
+            }
+            if(globalJson == null || globalJson.sdk == null || globalJson.sdk.version == null){
+                throw tl.error("FailedToReadGlobalJson \r\n" +  
+                "The global.json file isn't valid. (" + filePath + ") Please take a look in the documentation for global.json (https://docs.microsoft.com/en-us/dotnet/core/tools/global-json)");
+            }
+            sdkVersionNumber.push({ name: globalJson.sdk.version, toolPath: null });            
+        }
+        // check each version if it installed in the cache
+        for (let index = 0; index < sdkVersionNumber.length; index++) {
+            const d = sdkVersionNumber[index];
+            let path = this.getLocalTool(d.name);
+            d.toolPath = path;
+        }      
+        // download all sdk that are not in the cache
+        let sdksToDownload = sdkVersionNumber
+            // remove all tools that are present in cache.
+            .filter(function (d) {return d.toolPath == null})
+            // distinct on version name
+            .filter(function (x, i, a) {
```

Comment:
kindly use better temp variable names

Reply:
done.

Reply:
I think I have resolve this. If not reopen this conversion.